homeassistant/components/shelly/repairs.py

```diff
@@ -0,0 +1,77 @@
+"""Repairs flow for Shelly."""
+
+from __future__ import annotations
+
+from aioshelly.exceptions import DeviceConnectionError, RpcCallError
+from aioshelly.rpc_device import RpcDevice
+import voluptuous as vol
+
+from homeassistant import data_entry_flow
+from homeassistant.components.repairs import ConfirmRepairFlow, RepairsFlow
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import issue_registry as ir
+
+
+class BleScannerFirmwareUpdateFlow(RepairsFlow):
+    """Handler for an issue fixing flow."""
+
+    _device: RpcDevice
+
+    def __init__(self, device: RpcDevice) -> None:
+        """Initialize."""
+        self._device = device
+
+    async def async_step_init(
+        self, user_input: dict[str, str] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the first step of a fix flow."""
+        return await self.async_step_confirm()
+
+    async def async_step_confirm(
+        self, user_input: dict[str, str] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the confirm step of a fix flow."""
+        if user_input is not None:
+            return await self.async_step_update_firmware()
+
+        issue_registry = ir.async_get(self.hass)
+        description_placeholders = None
+        if issue := issue_registry.async_get_issue(self.handler, self.issue_id):
+            description_placeholders = issue.translation_placeholders
+
+        return self.async_show_form(
+            step_id="confirm",
+            data_schema=vol.Schema({}),
+            description_placeholders=description_placeholders,
+        )
+
+    async def async_step_update_firmware(
+        self, user_input: dict[str, str] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the confirm step of a fix flow."""
+        try:
+            await self._device.trigger_ota_update()
+        except RpcCallError as err:
+            if "Resource unavailable: No update info!" in str(err):
+                return self.async_abort(reason="update_not_available")
```

Comment:
Not for this PR, but in the future it would be nice if we had a named exception or this was abstracted away in the library

Reply:
Maybe checking for -114 is possible?

Reply:
I think instead of trying to trigger update an get an exception we can check if the device has update (same as we do in the update entity) if it doesn't we can abort

Reply:
Good idea